crates/re_viewport/src/viewport_blueprint.rs

```diff
@@ -0,0 +1,294 @@
+    // store, this will end up with default-visibility, which then *won't* be saved back.
+    for (_, view) in unknown_space_views {
+        viewport.add_space_view(view);
+    }
+
+    viewport
+}
+
+// ----------------------------------------------------------------------------
+
+fn sync_space_view(
+    deltas: &mut Vec<DataRow>,
+    space_view: &SpaceViewBlueprint,
+    snapshot: Option<&SpaceViewBlueprint>,
+) {
+    if Some(space_view) != snapshot {
+        let entity_path = EntityPath::from(format!(
+            "{}/{}",
+            SpaceViewComponent::SPACEVIEW_PREFIX,
+            space_view.id
+        ));
+
+        // TODO(jleibs): Seq instead of timeless?
+        let timepoint = TimePoint::timeless();
+
+        let component = SpaceViewComponent {
+            space_view: space_view.clone(),
+        };
+
+        push_one_component(deltas, &entity_path, &timepoint, component);
+    }
+}
+
+fn clear_space_view(deltas: &mut Vec<DataRow>, space_view_id: &SpaceViewId) {
+    let entity_path = EntityPath::from(format!(
+        "{}/{}",
+        SpaceViewComponent::SPACEVIEW_PREFIX,
+        space_view_id
+    ));
+
+    // TODO(jleibs): Seq instead of timeless?
+    let timepoint = TimePoint::timeless();
+
+    let cell =
+        DataCell::from_arrow_empty(SpaceViewComponent::name(), SpaceViewComponent::data_type());
+
+    let mut row = DataRow::from_cells1(RowId::random(), entity_path, timepoint, 0, cell);
+    row.compute_all_size_bytes();
+
+    deltas.push(row);
+}
+
+fn sync_viewport(deltas: &mut Vec<DataRow>, viewport: &Viewport, snapshot: &Viewport) {
+    let entity_path = EntityPath::from(VIEWPORT_PATH);
+
+    // TODO(jleibs): Seq instead of timeless?
+    let timepoint = TimePoint::timeless();
+
+    if viewport.auto_space_views != snapshot.auto_space_views {
+        let component = AutoSpaceViews(viewport.auto_space_views);
+        push_one_component(deltas, &entity_path, &timepoint, component);
+    }
+
+    if viewport.visible != snapshot.visible {
+        let component = SpaceViewVisibility(viewport.visible.clone());
+        push_one_component(deltas, &entity_path, &timepoint, component);
+    }
+
+    if viewport.maximized != snapshot.maximized {
+        let component = SpaceViewMaximized(viewport.maximized);
+        push_one_component(deltas, &entity_path, &timepoint, component);
+    }
+
+    // Note: we can't just check `viewport.trees != snapshot.trees` because the
+    // tree contains serde[skip]'d state that won't match in PartialEq.
+    if viewport.trees.len() != snapshot.trees.len()
+        || !viewport.trees.iter().zip(snapshot.trees.iter()).all(
+            |((left_vis, left_tree), (right_vis, right_tree))| {
+                left_vis == right_vis
+                    && left_tree.root == right_tree.root
+                    && left_tree.tiles.tiles == right_tree.tiles.tiles
+            },
+        )
+        || viewport.has_been_user_edited != snapshot.has_been_user_edited
+    {
```

Comment:
then shouldn't we manually implement `PartialEq` for whatever type is `viewport.trees` rather than defining this here?

Reply:
The general question of whether PartialEq should skip serde-skip fields is challenging. Especially for an external data-structure such as egui-tile-tree.